pkg/testhelper/testhelper.go

```diff
@@ -33,3 +38,36 @@ func ResultToString(result int) (str string) {
 	}
 	return ""
 }
+
+func SkipIfEmptyAny(skip func(string, ...int), object ...interface{}) {
+	for _, o := range object {
+		s := reflect.ValueOf(o)
+		if s.Kind() != reflect.Slice && s.Kind() != reflect.Map {
+			panic("SkipIfEmpty given a non-slice type")
+		}
+
+		if s.Len() == 0 {
+			skip(fmt.Sprintf("Test failed because there are no %s to test, please check under test labels", reflect.TypeOf(o)))
+		}
+	}
+}
+
+func SkipIfEmptyAll(skip func(string, ...int), object ...interface{}) {
+	countLenZero := 0
+	allTypes := ""
+	for _, o := range object {
+		s := reflect.ValueOf(o)
+		if s.Kind() != reflect.Slice && s.Kind() != reflect.Map {
+			panic("SkipIfEmpty given a non-slice type")
```

Comment:
non-slice and non-map.
once we get familiar with 1.18,  we can change this.
you can actually add constraint of the interface

Reply:
fixed